llvm/lib/SYCLLowerIR/LowerKernelProps.cpp

```diff
@@ -44,17 +44,31 @@ void processSetKernelPropertiesCall(CallInst &CI) {
 
   switch (PropID) {
   case property_ids::use_large_grf:
+  case property_ids::use_auto_grf:
     // TODO: Keep track of traversed functions to avoid repeating traversals
     // over same function.
-    llvm::sycl::utils::traverseCallgraphUp(F, [](Function *GraphNode) {
-      GraphNode->addFnAttr(::sycl::kernel_props::ATTR_LARGE_GRF);
-      // Add RegisterAllocMode metadata with arg 2 to the kernel to tell
-      // IGC to compile this kernel in large GRF mode. 2 means large.
+    llvm::sycl::utils::traverseCallgraphUp(F, [=](Function *GraphNode) {
+      auto AttrToAdd = PropID == property_ids::use_large_grf
+                           ? ::sycl::kernel_props::ATTR_LARGE_GRF
+                           : ::sycl::kernel_props::ATTR_AUTO_GRF;
+      if ((AttrToAdd == ::sycl::kernel_props::ATTR_LARGE_GRF &&
+           GraphNode->hasFnAttribute(::sycl::kernel_props::ATTR_AUTO_GRF)) ||
+          (AttrToAdd == ::sycl::kernel_props::ATTR_AUTO_GRF &&
+           GraphNode->hasFnAttribute(::sycl::kernel_props::ATTR_LARGE_GRF))) {
+        llvm::report_fatal_error(llvm::Twine(__FILE__ " ") +
```

Comment:
`__FILE__` here resolves to `LowerKernelProps.cpp`. It wouldn't be of use for a customer - and doesn't help to find which exact kernel is problematic. I would recommend implementing a header-level diagnostic, if possible

Reply:
yeah this is gonna get removed, thanks